kimu/core/rectangular_tool.py

```diff
@@ -0,0 +1,403 @@
+
+        # Coordinates of the first solution point
+        x_b1 = float(
+            (line_coords[3] * b_measure * Decimal(math.sqrt(
+                a2 ** Decimal("2.0") + b2 ** Decimal("2.0")
+            ))
+              - line_coords[1] * b_measure * Decimal(math.sqrt(
+                    a2 ** Decimal("2.0") + b2 ** Decimal("2.0")
+                )) - b2 * y_a * line_coords[3]
+              + b2 * y_a * line_coords[1]
+              - b2 * x_a * line_coords[2] + b2 * x_a * line_coords[0]
+              - c2 * line_coords[3] + c2 * line_coords[1]
+              )
+            / (a2 * line_coords[3] - a2 * line_coords[1]
+               - b2 * line_coords[2] + b2 * line_coords[0])
+        )
+
+        y_b1 = float(
+            (y_a * line_coords[3] - y_a * line_coords[1]
+             - Decimal(x_b1) * line_coords[2]
+             + Decimal(x_b1) * line_coords[0]
+             + x_a * line_coords[2] - x_a * line_coords[0])
+            / (line_coords[3] - line_coords[1])
+        )
+
+        # Coordinates of the second solution point
+        x_b2 = float(
+            (-line_coords[3] * b_measure * Decimal(math.sqrt(
+                a2 ** Decimal("2.0") + b2 ** Decimal("2.0")
+            ))
+             + line_coords[1] * b_measure * Decimal(math.sqrt(
+                    a2 ** Decimal("2.0") + b2 ** Decimal("2.0")
+                )) - b2 * y_a * line_coords[3]
+             + b2 * y_a * line_coords[1]
+             - b2 * x_a * line_coords[2] + b2 * x_a * line_coords[0]
+             - c2 * line_coords[3] + c2 * line_coords[1]
+             )
+            / (a2 * line_coords[3] - a2 * line_coords[1]
+               - b2 * line_coords[2] + b2 * line_coords[0])
+        )
+
+        y_b2 = float(
+            (y_a * line_coords[3] - y_a * line_coords[1]
+             - Decimal(x_b2) * line_coords[2]
+             + Decimal(x_b2) * line_coords[0]
+             + x_a * line_coords[2] - x_a * line_coords[0])
+            / (line_coords[3] - line_coords[1])
+        )
+
+        # Check that the intersection point(s) lie(s) in the
+        # map canvas extent
+        extent = iface.mapCanvas().extent()
+
+        if (
+            x_b1 < extent.xMinimum()
+            or x_b1 > extent.xMaximum()
+            or y_b1 < extent.yMinimum()
+            or y_b1 > extent.yMaximum()
+        ):
+            LOGGER.warning(
+                tr("Solution point 1 lies outside of the map canvas!"),
+                extra={"details": ""},
+            )
+            return
+
+        if (
+            x_b2 < extent.xMinimum()
+            or x_b2 > extent.xMaximum()
+            or y_b2 < extent.yMinimum()
+            or y_b2 > extent.yMaximum()
+        ):
+            LOGGER.warning(
+                tr("Solution point 2 lies outside of the map canvas!"),
+                extra={"details": ""},
+            )
+            return
+
+        # Add result layers to map canvas
+        self._add_result_layers(x_b1, y_b1, x_b2, y_b2)
```

Comment:
The method name `_locate_point_b` does not imply that any result layers are added, so this is likely an unexpected side-effect for those who call this method. Solution: simply return the point coordinates, and call _add_result_layers elsewhere (i.e. in the main canvaspressevent method).